### PR TITLE
Feature: Combine metric+palette into metricSpec (#118)

### LINKED ISSUE #118: Feature: Command line upgrade
To simplify the command line, and to make things easier to read, we want to combine multiple parameters for a single vizualization feature into one.

Instead of having a pair of parameters, one for the metric and one for the palette, we want to have a single parameter that captures both.

For example, instead of 
``` bash
--border file-lines --border-palette foliage --fill file-type --fill-palette categorization
```

We want to have
```
--border filelines,foliage --fill file-type,categorization
```

This will also preposition us for future extensions to metrics as we can add a third parameter to the existing structure.

## Implementation Guide

* Introduce a new type `metricSpec` in the `config` package to capture both `Metric` and `Palette`.
* Use `metricSpec` for both the command line, and for configuration.
* Kong integration by implementing either `encoding.TextUnmarshaler` or `kong.MapperValue`.

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
Fixes #118

## Summary

Introduces `config.MetricSpec` type that bundles a metric name and palette name into a single CLI parameter. This simplifies the command line:

**Before:**
```
--fill file-type --fill-palette categorization --border file-lines --border-palette foliage
```

**After:**
```
--fill file-type,categorization --border file-lines,foliage
```

## Changes

- **New `MetricSpec` type** (`internal/config/metric_spec.go`): Captures both metric and palette. Implements `encoding.TextUnmarshaler` for Kong CLI parsing, plus YAML/JSON marshal/unmarshal for config files.
- **Updated CLI structs**: `TreemapCmd`, `RadialCmd`, `BubbletreeCmd` now use `MetricSpec` for `--fill` and `--border` flags instead of separate metric + palette parameters.
- **Updated config structs**: `Treemap`, `Radial`, `Bubbletree` use `*MetricSpec` for Fill and Border, replacing separate `Fill`/`FillPalette`/`Border`/`BorderPalette` pointer strings.
- **Removed flags**: `--fill-palette` and `--border-palette` are no longer separate flags.

## Config file format change

Config files now use the combined format:
```yaml
treemap:
  fill: file-type,categorization
  border: file-lines,foliage
```

Metric-only (no explicit palette) still works — the provider default palette is used:
```yaml
treemap:
  fill: file-type
```

## Breaking changes

- `--fill-palette` and `--border-palette` CLI flags removed (use `--fill metric,palette` instead).
- Config file fields `fillPalette` and `borderPalette` removed (combine into `fill` and `border` values).
- The `--border-palette requires --border` validation is no longer needed (palette is always bundled with its metric).

## Tests

22 new tests for MetricSpec (parsing, round-trips, edge cases, YAML/JSON). All existing tests updated and passing.

### CODE REVIEW

File: internal/config/metric_spec.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	"go.yaml.in/yaml/v3"
+)
+
+// MetricSpec combines a metric name and an optional palette name into a single
+// value. It implements encoding.TextUnmarshaler so Kong can parse "metric,palette"
+// from the command line, and custom YAML/JSON marshaling for config files.
+//
+// Format: "metric" or "metric,palette".
+type MetricSpec struct { //nolint:recvcheck // marshal methods need value receivers, unmarshal need pointer
+	Metric  string
```
Comment (theunrepentantgeek) [inline internal/config/metric_spec.go]: This shouldn't be a string, it should be a `metric.Name` to save all those onerous casts.

File: internal/config/metric_spec.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	Palette string
```
Comment (theunrepentantgeek) [inline internal/config/metric_spec.go]: This should be a `palette.Name` for the same reason.

File: internal/config/metric_spec.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+// Implements encoding.TextUnmarshaler for Kong CLI integration.
+func (m *MetricSpec) UnmarshalText(text []byte) error {
+	s := strings.TrimSpace(string(text))
+	if s == "" {
+		*m = MetricSpec{}
+
+		return nil
+	}
+
+	parts := strings.SplitN(s, ",", 2)
```
Comment (theunrepentantgeek) [inline internal/config/metric_spec.go]: Poor choice, as this will both do the wrong thing if there are more than two parts given by the user, and require significant reworking when a third (or fourth) part is supported. The later was previously flagged as coming and should have been taken into account.

Use `strings.Cut()` to pull out the metric, handle it, and then use `strings.Cut()` a second time to handle the palette. Return an error if there's anything left after that point. Simpler to understand, and easily extensible in the future.

File: cmd/codeviz/bubbletree_cmd.go [hunk 4]
```diff
@@ -75,19 +66,11 @@ func (*BubbletreeCmd) validateConfig(cfg *config.Bubbletree) error {
 		return eris.Errorf("size metric must be numeric, got %q (kind: %d)", size, p.Kind())
 	}
 
-	if err := validateMetricPalette(ptrString(cfg.Fill), ptrString(cfg.FillPalette), "fill"); err != nil {
+	if err := validateMetricPalette(specMetric(cfg.Fill), specPalette(cfg.Fill), "fill"); err != nil {
```
Comment (theunrepentantgeek) [inline cmd/codeviz/bubbletree_cmd.go]: The _Feature Envy_ design smell applies - `validateMetricPalette()` should be refactored to be a method on our new `MetricSpec` type, accepting just the label. Calling it will look much tidier:

```suggestion
	if err := cfg.Fill.Validate("fill"); err != nil {
```

File: internal/config/metric_spec.go (hunk 3 above)
Comment (theunrepentantgeek) [reply internal/config/metric_spec.go]: Done. Replaced `strings.SplitN` with two `strings.Cut()` calls — first to extract the metric, second to extract the palette. Returns an error if there's any extra content after the palette. Updated the corresponding test to expect an error on `"file-type,a,b"`.

File: internal/config/metric_spec.go (hunk 2 above)
Comment (theunrepentantgeek) [reply internal/config/metric_spec.go]: Done. Changed `Palette` field to `palette.PaletteName`. Casts eliminated throughout.

File: internal/config/metric_spec.go (hunk 1 above)
Comment (theunrepentantgeek) [reply internal/config/metric_spec.go]: Done. Changed `Metric` field to `metric.Name`. All downstream casts removed across treemap, radial, and bubbletree command files.

File: cmd/codeviz/bubbletree_cmd.go (hunk 4 above)
Comment (theunrepentantgeek) [reply cmd/codeviz/bubbletree_cmd.go]: Done. Moved validation into `MetricSpec.Validate(label string) error` with nil-safe pointer receiver. All three command files (treemap, radial, bubbletree) now call `cfg.Fill.Validate("fill")` and `cfg.Border.Validate("border")`. The standalone `validateMetricPalette()`, `specMetric()`, and `specPalette()` helpers simplified — specMetric/specPalette now return typed `metric.Name`/`palette.PaletteName`.
